programs/drift_vaults/src/instructions/initialize_vault.rs

```diff
@@ -150,19 +205,181 @@ impl<'info> InitializeUserCPI for Context<'_, '_, '_, 'info, InitializeVault<'in
         let signers = &[&signature_seeds[..]];
 
         let cpi_program = self.accounts.drift_program.to_account_info().clone();
-        let cpi_accounts = InitializeUser {
+        let cpi_accounts = InitializeUserStats {
             user_stats: self.accounts.drift_user_stats.clone(),
-            user: self.accounts.drift_user.clone(),
             state: self.accounts.drift_state.clone(),
             authority: self.accounts.vault.to_account_info().clone(),
             payer: self.accounts.payer.to_account_info().clone(),
             rent: self.accounts.rent.to_account_info().clone(),
             system_program: self.accounts.system_program.to_account_info().clone(),
         };
         let cpi_ctx = CpiContext::new_with_signer(cpi_program, cpi_accounts, signers);
-        let sub_account_id = 0_u16;
-        drift::cpi::initialize_user(cpi_ctx, sub_account_id, name)?;
+        drift::cpi::initialize_user_stats(cpi_ctx)?;
 
         Ok(())
     }
 }
+
+// this function is needed because remaining accounts can't have account macro constraints,
```

Comment:
dont love that we manually copy over the anchor code, what do you think of just adding a initialize_vault_with_protocol?

Reply:
That'd be cleaner, yes.